src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -70,12 +68,26 @@ private Task PackageInitializationBackgroundThreadAsync(PackageLoadTasks package
             // Ensure we're on the BG when creating the language service.
             await TaskScheduler.Default;
 
-            // Create the language service, tell it to set itself up, then store it in a field
-            // so we can notify it that it's time to clean up.
-            _languageService = CreateLanguageService();
-            await _languageService.SetupAsync(cancellationToken).ConfigureAwait(false);
-
-            return _languageService.ComAggregate!;
+            var languageService = CreateLanguageService();
+            languageService.Setup(cancellationToken);
+
+            // DevDiv 753309:
+            // We've redefined some VS interfaces that had incorrect PIAs. When 
+            // we interop with native parts of VS, they always QI, so everything
+            // works. However, Razor is now managed, but assumes that the C#
+            // language service is native. When setting breakpoints, they
+            // get the language service from its GUID and cast it to IVsLanguageDebugInfo.
+            // This would QI the native lang service. Since we're managed and
+            // we've redefined IVsLanguageDebugInfo, the cast
+            // fails. To work around this, we put the LS inside a ComAggregate object,
+            // which always force a QueryInterface and allow their cast to succeed.
+            // 
+            // This also fixes 752331, which is a similar problem with the 
+            // exception assistant.
+
+            // Creating the com aggregate has to happen on the UI thread.
+            await this.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            return Interop.ComAggregate.CreateAggregatedObject(languageService);
```

Comment:
Could this be done by utilizing packageInitializationTasks.AddTask instead of doing an extra main thread switch?

Reply:
We don't actually create the object until it's requested though, so aren't using that initialization pattern here.